sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchSynonymMapAsyncClient.java

```diff
@@ -28,24 +29,32 @@
  * Asynchronous Client to manage and query synonym map, as well as manage other resources,
  * on a Cognitive Search service.
  */
-@ServiceClient(builder = SearchServiceResourceClientBuilder.class, isAsync = true)
-public class SearchSynonymMapAsyncClient extends SearchServiceResourceAsyncClientBase {
+public class SearchSynonymMapAsyncClient {
```

Comment:
We weren't renaming this class. `SynonymMap` had no complaints about being too generic. This wasn't part of the rename issue: #10272.

Everything related to the model name (client, methods, etc.) should be fixed.